src/sdk/conference/channel.js

```diff
@@ -77,20 +77,20 @@ export class ConferencePeerConnectionChannel extends EventDispatcher {
 
   publish(stream, options) {
     if (options === undefined) {
-      options = {audio: !!stream.mediaStream.getAudioTracks(), video: !!stream
-          .mediaStream.getVideoTracks()};
+      options = {audio: !!stream.mediaStream.getAudioTracks().length, video: !!stream
+          .mediaStream.getVideoTracks().length};
     }
     if (typeof options !== 'object') {
       return Promise.reject(new TypeError('Options should be an object.'));
     }
     if (options.audio === undefined) {
-      options.audio = !!stream.mediaStream.getAudioTracks();
+      options.audio = !!stream.mediaStream.getAudioTracks().length;
     }
     if (options.video === undefined) {
-      options.video = !!stream.mediaStream.getVideoTracks();
+      options.video = !!stream.mediaStream.getVideoTracks().length;
     }
-    if (!!options.audio === !stream.mediaStream.getAudioTracks().length || !!
-    options.video === !stream.mediaStream.getVideoTracks().length) {
+    if (!!options.audio > !!stream.mediaStream.getAudioTracks().length || !!
```

Comment:
Comparing two boolean values by > looks a little bit wired.

Reply:
I also think so, but if not use this, the condition for if statement will be:
if ( (!!options.audio == true && !!stream.mediaStream.getAudioTracks().length == false) || 
(!!options.video == true && !!stream.mediaStream.getVideoTracks().length == false) )
I think it is much too long and not so well.

Reply:
add comment to this condition